src/vs/platform/gpu/electron-main/gpuStatusMainService.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { BrowserWindow } from 'electron';
+import { IGPUStatusMainService, OpenGPUStatusWindowOptions } from '../common/gpuStatusMain.js';
+import { IEnvironmentMainService } from '../../environment/electron-main/environmentMainService.js';
+import { zoomLevelToZoomFactor } from '../../window/common/window.js';
+
+export class GPUStatusMainService implements IGPUStatusMainService {
+	declare readonly _serviceBrand: undefined;
+	private gpuStatusWindow: BrowserWindow | null = null;
+	private gpuStatusParentWindow: BrowserWindow | null = null;
+
+	constructor(
+		@IEnvironmentMainService private readonly _environmentMainService: IEnvironmentMainService,
+	) { }
+
+	openGPUStatusWindow(options: OpenGPUStatusWindowOptions) {
+		if (!this.gpuStatusWindow) {
+			this.gpuStatusParentWindow = BrowserWindow.getFocusedWindow();
+			if (this.gpuStatusParentWindow) {
+				this.gpuStatusWindow = this.createBrowserWindow({ zoomLevel: options.zoomLevel });
```

Comment:
nit: window doesn't seem centered 🤔

Reply:
> instead of creating a BrowserWindow object, maybe consider taking a look at utilizing the Auxiliary Window Service (via auxiliaryWindowService.open), which will give you the built in services and may be a little cleaner vs. creating a new BrowserWindow.

Use auxiliaryWindowService makes sense, previously `issue report` used `BrowserWindow` directly so I follow the same approach before 😄.

I take a look at how `auxiliaryWindowService.open` works it leverages `window.open` to open a blank window with `about:blank`.

But what we need here is an internal page load with `chrome://gpu` it's an internal protocol for Chrome. `window.open` just doesn't work in this situation.

pseudocode

```ts
async openGPUStatusWindow(options: OpenGPUStatusWindowOptions) {
		const disposables = new DisposableStore();
		const auxiliaryWindow = disposables.add(await this._auxiliaryWindowService.open({
			mode: AuxiliaryWindowMode.Normal,
			bounds: { width: 800, height: 600 },
			zoomLevel: options.zoomLevel,
			nativeTitlebar: true,
			disableFullscreen: true
		}));

		if (auxiliaryWindow) {
			auxiliaryWindow.window.document.title = 'GPU Status';
			auxiliaryWindow.window.addEventListener('error', (e) => {
				console.error('GPU Status Window Error', e);
			});
			auxiliaryWindow.window.location.href = 'chrome://gpu';
		} else {
			console.error('Failed to open auxiliary window');
		}
	}
```

> nit: window doesn't seem centered 🤔

I tested it on macOS seems to open in the center of the monitor every time. Proberly different in other platform maybe we should add `center: true`  in `BrowserWindowOptions`

Reply:
> But what we need here is an internal page load with chrome://gpu it's an internal protocol for Chrome. window.open just doesn't work in this circumstance.

`window.open` will support chrome webui pages, the issue here is the process lock. Aux windows are synchronous windows that share the same process as the opener window and in this case the process is locked to the origin `vscode-file://vscode-app`, however loading `chrome://gpu` will require the process to be locked to this webui pages origin and that is not possible. Hence the runtime will bail at this [condition](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/child_process_security_policy_impl.cc;l=1304-1319) and replace the load with `about:blank#blocked`.

Reply:
I am good with using a separate BrowserWindow but maybe we rename the commands to `Developer: Show GPU Info` and remove the menu entries to keep it less exposed for debugging purposes only.

Reply:
> > But what we need here is an internal page load with chrome://gpu it's an internal protocol for Chrome. window.open just doesn't work in this circumstance.
> 
> `window.open` will support chrome webui pages, the issue here is the process lock. Aux windows are synchronous windows that share the same process as the opener window and in this case the process is locked to the origin `vscode-file://vscode-app`, however loading `chrome://gpu` will require the process to be locked to this webui pages origin and that is not possible. Hence the runtime will bail at this [condition](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/child_process_security_policy_impl.cc;l=1304-1319) and replace the load with `about:blank#blocked`.

Cool. TIL!